relay-server/src/utils/rate_limits.rs

```diff
@@ -573,10 +580,14 @@ where
                 }
 
                 enforcement.event = CategoryLimit::new(index_category, 1, longest);
+                enforcement.spans =
+                    CategoryLimit::new(DataCategory::SpanIndexed, summary.span_quantity, longest);
             } else {
                 event_limits = (self.check)(scoping.item(category), 1)?;
                 longest = event_limits.longest();
                 enforcement.event = CategoryLimit::new(category, 1, longest);
+                enforcement.spans =
+                    CategoryLimit::new(DataCategory::Span, summary.span_quantity, longest);
```

Comment:
nit: I would prefer if we put all the new logic in the `if summary.span_quantity > 0` block so that it's easy to find when we look for span-related logic.

Reply:
I did do that in my first iteration but removed because of verbosity and because a category is inactive if it's count is 0 but I can add it back.